kover-gradle-plugin/src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/internal/ReportsImpl.kt

```diff
@@ -0,0 +1,354 @@
+    }
+
+    override fun variant(variant: String, config: Action<KoverReportVariantConfig>) {
+        val report = byName.getOrPut(variant) {
+            createVariant(variant)
+        }
+        config(report)
+    }
+
+    internal fun createVariant(variantName: String): KoverReportVariantConfigImpl {
+        val block = objects.newInstance<KoverReportVariantConfigImpl>(objects, layout.buildDirectory, variantName)
+
+        block.filters.extendsFrom(rootFilters)
+        block.verify.extendFrom(rootVerify)
+
+        return block
+    }
+}
+
+internal abstract class KoverReportVariantConfigImpl @Inject constructor(
+    objects: ObjectFactory,
+    buildDir: DirectoryProperty,
+    private val variantName: String
+) : KoverReportVariantConfig {
+    internal val filters: KoverReportFiltersConfigImpl = objects.newInstance()
+    internal val verify: KoverVerifyTaskConfigImpl = objects.newInstance()
+
+    internal val html: KoverHtmlTaskConfigImpl = objects.newInstance()
+    internal val xml: KoverXmlTaskConfigImpl = objects.newInstance()
+    internal val binary: KoverBinaryTaskConfigImpl = objects.newInstance()
+    internal val log: KoverLogTaskConfigImpl = objects.newInstance()
+
+    init {
+        xml.xmlFile.convention(buildDir.file(xmlReportPath(variantName)))
+        html.htmlDir.convention(buildDir.dir(htmlReportPath(variantName)))
+        binary.file.convention(buildDir.file(binaryReportPath(variantName)))
+
+        html.onCheck.convention(false)
+        xml.onCheck.convention(false)
+        binary.onCheck.convention(false)
+        log.onCheck.convention(false)
+        verify.onCheck.convention(variantName == TOTAL_VARIANT_NAME)
+
+
+        log.format.convention("<entity> line coverage: <value>%")
+        log.groupBy.convention(GroupingEntityType.APPLICATION)
+        log.coverageUnits.convention(MetricType.LINE)
+        log.aggregationForGroup.convention(AggregationType.COVERED_PERCENTAGE)
+    }
+
+
+    override fun filters(config: Action<KoverReportFiltersConfig>) {
+        filters.clean()
+        config(filters)
+    }
+
+    override fun filtersAppend(config: Action<KoverReportFiltersConfig>) {
+        config(filters)
+    }
+
+    override fun html(config: Action<KoverHtmlTaskConfig>) {
+        config(html)
+    }
+
+    override fun xml(config: Action<KoverXmlTaskConfig>) {
+        config(xml)
+    }
+
+    override fun binary(config: Action<KoverBinaryTaskConfig>) {
+        config(binary)
+    }
+
+    override fun verify(config: Action<KoverVerifyTaskConfig>) {
+        verify.clean()
+        config(verify)
+    }
+
+    override fun verifyAppend(config: Action<KoverVerifyTaskConfig>) {
+        config(verify)
+    }
+
+    override fun log(config: Action<KoverLogTaskConfig>) {
+        config(log)
+    }
+}
+
+internal abstract class KoverHtmlTaskConfigImpl : KoverHtmlTaskConfig {
+    override fun filters(config: Action<KoverReportFiltersConfig>) {
+        TODO("deprecation")
```

Comment:
I'd instead write here a full message, e.g., `This method is deprecated and shouldn't be used`.